includes/core/server/request/class-create-and-confirm-setup-intention.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Class file for WCPay\Core\Server\Request\Create_Setup_Intention.
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core\Server\Request;
+
+use WC_Payments;
+use WCPay\Core\Exceptions\Server\Request\Invalid_Request_Parameter_Exception;
+use WCPay\Core\Server\Request;
+use WC_Payments_API_Client;
+
+/**
+ * Request class for creating setup intents.
+ */
+class Create_And_Confirm_Setup_Intention extends Request {
+	use Intention;
+	use Level3;
+
+	const IMMUTABLE_PARAMS = [ 'customer', 'confirm' ];
+	const REQUIRED_PARAMS  = [ 'customer' ];
+
+	const DEFAULT_PARAMS = [
+		'confirm'  => 'true',
+		'metadata' => [],
+	];
+
+	/**
+	 * Returns the request's API.
+	 *
+	 * @return string
+	 */
+	public function get_api(): string {
+		return WC_Payments_API_Client::SETUP_INTENTS_API;
+	}
+
+	/**
+	 * Returns the request's HTTP method.
+	 */
+	public function get_method(): string {
+		return 'POST';
+	}
+
+	/**
+	 * Customer setter.
+	 *
+	 * @param string $customer_id ID of the customer making the payment.
+	 * @return void
+	 *
+	 * @throws Invalid_Request_Parameter_Exception
+	 */
+	public function set_customer( string $customer_id ) {
+		$this->validate_stripe_id( $customer_id, 'cus' );
+		$this->set_param( 'customer', $customer_id );
+	}
+
+	/**
+	 * Set metadata.
+	 *
+	 * @param array $metadata Metadata to set.
+	 *
+	 * @return void
+	 */
+	public function set_metadata( array $metadata ) {
+		$this->set_param( 'metadata', $metadata );
```

Comment:
I'm wondering whether we should add similar processing here to `Create_Intention`'s `set_metadata` method, namely the intent description. I know it was like this before, but IMO that's a missed opportunity rather than done intentionally.

Reply:
I am sorry, but I am not 100% certain what do you mean here. Do you want to use same metadata setter with description and fingerprint here from `Create_Intention` class or do you want to use simple metadata setter where you don't add description and fingerprinting to `Create_Intention` request class?

Adding description is easy, and we could include it easily. Please let me know what you had in mind and I can easily implement it.